### PR TITLE
[ADD] Extra visualization example

### PR DESCRIPTION
**Pull request by franchuterivera**
In preparation to move the refactor_development branch to be development, we were missing:

* User details in the documentation
* Examples with visualization

This PR aims to fix this.

### CODE REVIEW

File: docs/manual.rst [hunk 1]
```diff
@@ -6,4 +6,43 @@
 Manual
 ======
 
-TODO
+This manual shows how to get started with Auto-PyTorch. We recommend going over the examples first.
+There are additional recommendations on how to interact with the API, further below in this manual.
+However, you are welcome to contribute to this documentation but starting a Pull-Request.
```
Comment (ravinkohli) [inline docs/manual.rst:11]: by making a Pull Request?

File: docs/manual.rst [hunk 2]
```diff
@@ -6,4 +6,43 @@
...
+* `Customizing the search space <examples/tabular/40_advanced/example_custom_configuration_space.html>`_
+* `Changing the resampling strategy <examples/tabular/40_advanced/example_resampling_strategy.html>`_
+* `Visualizing the results <examples/tabular/40_advanced/example_visualization.html>`_
+
+Resource Allocation
+===================
+
+Auto-PyTorch allows to control the maximum allowed resident set memory that an estimator can use. By providing the `memory_limit` argument to the `search()` method, one can make sure that neither the individual machine learning models fitted by SMAC nor the final ensemble consume more than `memory_limit` megabytes.
+
+Additionally, one can control the allocated time to search for a model, via the argument `total_walltime_limit` to the `search()` method. The later controls how much time SMAC can search for new configurations to solve the problem at hand. The more time is allocated, the better the final estimator will be.
```
Comment (ravinkohli) [inline docs/manual.rst:28]: the latter* controls

Comment (ravinkohli) [reply docs/manual.rst:28]: Actually I dont think you need to put latter here. it seems like you are talking about the search method

File: docs/manual.rst (hunk 1 above)
Comment (ravinkohli) [inline docs/manual.rst:11]:
```suggestion
However, you are welcome to contribute to this documentation by making a Pull-Request.
```

File: docs/manual.rst (hunk 2 above)
Comment (ravinkohli) [inline docs/manual.rst:28]:
```suggestion
Additionally, one can control the allocated time to search for a model, via the argument `total_walltime_limit` to the `search()` method. This argument controls how much time SMAC can search for new configurations to solve the problem at hand. The more time is allocated, the better the final estimator will be.
```

**Review by ravinkohli — COMMENTED** [review]
Thank you for this PR. It shows vital features of the API and the extensive compatibility with sklearn API. I have made minor suggestions in the documentation but other than that, it can be merged

**Comment by franchuterivera** [comment]
Thanks a lot for the suggestions, I have committed them!

**Review by ravinkohli — APPROVED** [review]

**Comment by ravinkohli** [comment]
Actually could you also change the version [here](https://github.com/franchuterivera/Auto-PyTorch/blob/fb370c739ba0d2b593bedabc6b7ef3f7cf097d19/autoPyTorch/__version__.py#L4)

**Comment by ArlindKadra** [comment]
Really nice example :)

File: examples/tabular/40_advanced/example_visualization.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    pd.merge(
+        ensemble_performance_frame,
+        individual_performance_frame,
+        on="Timestamp", how='outer'
+    ).sort_values('Timestamp').fillna(method='ffill').plot(
+        x='Timestamp',
+        kind='line',
+        legend=True,
+        title='Auto-sklearn accuracy over time',
```
Comment (ravinkohli) [inline examples/tabular/40_advanced/example_visualization.py:148]:
```suggestion
        title='Auto-PyTorch accuracy over time',
```

File: docs/manual.rst [hunk 4]
```diff
@@ -6,4 +6,43 @@
...
+
+Ensemble Building Process
+=========================
+
+Auto-PyTorch uses ensemble selection by `Caruana et al. (2004) <https://dl.acm.org/doi/pdf/10.1145/1015330.1015432>`_
+to build an ensemble based on the models’ prediction for the validation set. The following hyperparameters control how the ensemble is constructed:
+
+* ``ensemble_size`` determines the maximal size of the ensemble. If it is set to zero, no ensemble will be constructed.
+* ``ensemble_nbest`` allows the user to directly specify the number of models considered for the ensemble.  This hyperparameter can be an integer *n*, such that only the best *n* models are used in the final ensemble. If a float between 0.0 and 1.0 is provided, ``ensemble_nbest`` would be interpreted as a fraction suggesting the percentage of models to use in the ensemble building process (namely, if ensemble_nbest is a float, library pruning is implemented as described in `Caruana et al. (2006) <https://dl.acm.org/doi/10.1109/ICDM.2006.76>`_).
```
Comment (nabenabe0928) [inline docs/manual.rst:37]: > This hyperparameter can be an integer n, such that only the best *n* models are used in the final ensemble

When an integer is provided for this hyperparameter, the final ensemble chooses each predictor from only the best n models

> If a float between 0.0 and 1.0 is provided, ``ensemble_nbest`` would be interpreted as a fraction suggesting the percentage of models to use in the ensemble building process

If a float between 0.0 and 1.0 is provided, ``ensemble_nbest`` will be interpreted as a fraction suggesting the percentage of models to use in the ensemble building process.
For example, when we train 10 models before the ensemble building process and the hyperparameter is 0.7, we build an ensemble by taking some models from only **best (?)** 7 models among 10 models.

File: docs/manual.rst [hunk 5]
```diff
@@ -6,4 +6,43 @@
...
+* ``max_models_on_disc`` defines the maximum number of models that are kept on the disc, as a mechanism to control the amount of disc space consumed by *auto-sklearn*. Throughout the automl process, different individual models are optimized, and their predictions (and other metadata) is stored on disc. The user can set the upper bound on how many models are acceptable to keep on disc, yet this variable takes priority in the definition of the number of models used by the ensemble builder (that is, the minimum of ``ensemble_size``, ``ensemble_nbest`` and ``max_models_on_disc`` determines the maximal amount of models used in the ensemble). If set to None, this feature is disabled.
```
Comment (nabenabe0928) [inline docs/manual.rst:38]: > the maximum number of models that are kept on the disc

the maximum number of models that can stay on the disc

> the amount of disc space consumed by *auto-sklearn*

auto-pytorch?

> their predictions (and other metadata) is stored on disc. 

are stored

File: examples/tabular/40_advanced/example_visualization.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
+"""
```
Comment (nabenabe0928) [inline examples/tabular/40_advanced/example_visualization.py:1]: Haven't checked yet

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]

File: examples/tabular/40_advanced/example_visualization.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+import os
+import pickle
+import tempfile as tmp
+import time
+import warnings
+
+os.environ['JOBLIB_TEMP_FOLDER'] = tmp.gettempdir()
+os.environ['OMP_NUM_THREADS'] = '1'
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
```
Comment (nabenabe0928) [inline examples/tabular/40_advanced/example_visualization.py:28]: What about those environment variables?
We need explanations for them.

**Review by nabenabe0928 — APPROVED** [review]

**Pull request merged**